src/ParallelTempering.jl

```diff
@@ -144,55 +306,189 @@ function run!(replica::Replica, T_sched::Function;
         end
     end
 
-    # set sampler's energy and magnetization values
+    # initialize energy and equilibrate replica to it's distribution
     replica.sampler.E = energy(replica.sampler.system)
     replica.sampler.M = sum(replica.sampler.system)
```

Comment:
See the other comment below -- please use `reset_running_energy!` and `reset_running_mag!`